src/main/java/gyro/azure/compute/VirtualMachineResource.java

```diff
@@ -868,7 +1131,13 @@ public void update(GyroUI ui, State state, Resource current, Set<String> changed
     public void delete(GyroUI ui, State state) {
         Azure client = createClient();
 
+        VirtualMachine virtualMachine = client.virtualMachines().getById(getId());
         client.virtualMachines().deleteById(getId());
+
+        if (isDeleteOsDiskOnTerminate()
+                && virtualMachine != null) {
+            client.disks().deleteById(virtualMachine.osDiskId());
```

Comment:
Should we delete the disk if it is managed ?

I feel if we have don't have the os disk configured then we delete the disk, otherwise leave it be